### PR TITLE
fix!: require cloud9 image_id to align with AWS breaking change

### LINKED ISSUE #34995: [Enhancement]: Cloud9 support Amazon Linux 2023
### Description

Cloud9 supports now Amazon Linux 2023 see [here](https://aws.amazon.com/about-aws/whats-new/2023/12/aws-cloud9-amazon-linux-2023/). 

Currently the [Cloud9 resource](aws_cloud9_environment_ec2) supports providing an image_id, which is passed to the API (cf. https://docs.aws.amazon.com/cloud9/latest/APIReference/API_CreateEnvironmentEC2.html). Unfortunately the Terraform Cloud9 resource does NOT know the new Amazon Linux 2023 images, although they are supported by the AWS API:
* amazonlinux-2023-x86_64
* resolve:ssm:/aws/service/cloud9/amis/amazonlinux-2023-x86_64

This means that you simply should allow also in Terraform those images.

### Affected Resource(s) and/or Data Source(s)

* aws_cloud9_environment_ec2

### Potential Terraform Configuration

```terraform
resource "aws_cloud9_environment_ec2" "example" {
  instance_type = "t2.micro"
  name          = "example-env"
  image_id      = "resolve:ssm:/aws/service/cloud9/amis/amazonlinux-2023-x86_64"
}
```

### References

https://docs.aws.amazon.com/cloud9/latest/APIReference/API_CreateEnvironmentEC2.html

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloud9_environment_ec2

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by JonZeolla**
### Description

The goal of this PR was to ad support for Cloud9 and Amazon Linux 2023, which was added on [Dec 18, 2023](https://aws.amazon.com/about-aws/whats-new/2023/12/aws-cloud9-amazon-linux-2023/).

In doing so, I noticed a few things, primarily that there was an upstream breaking change and this fixes cloud9_environment_ec2 resource provisioning:
1. `imageId` changed to required as of December 4, 2023 per [the docs](https://docs.aws.amazon.com/cloud9/latest/APIReference/API_CreateEnvironmentEC2.html#API_CreateEnvironmentEC2_RequestSyntax). This PR includes that **BREAKING** change as well.
2. `imageId` is not returned by any of the Cloud9 API endpoints, so we need to exclude it from state importing.
3. On January 22, 2024, Amazon Linux (AL1) will be removed from the list of available image IDs for Cloud9, per [the same docs](https://docs.aws.amazon.com/cloud9/latest/APIReference/API_CreateEnvironmentEC2.html#API_CreateEnvironmentEC2_RequestSyntax). I didn't make that change in this PR, since it hasn't happened yet.

### Relations

Closes #34995

### Output from Acceptance Testing

```console
➜ make testacc TESTS=TestAccCloud9EnvironmentEC2 PKG=cloud9
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/cloud9/... -v -count 1 -parallel 20 -run='TestAccCloud9EnvironmentEC2'  -timeout 360m
=== RUN   TestAccCloud9EnvironmentEC2_basic
=== PAUSE TestAccCloud9EnvironmentEC2_basic
=== RUN   TestAccCloud9EnvironmentEC2_allFields
=== PAUSE TestAccCloud9EnvironmentEC2_allFields
=== RUN   TestAccCloud9EnvironmentEC2_tags
=== PAUSE TestAccCloud9EnvironmentEC2_tags
=== RUN   TestAccCloud9EnvironmentEC2_disappears
=== PAUSE TestAccCloud9EnvironmentEC2_disappears
=== CONT  TestAccCloud9EnvironmentEC2_basic
=== CONT  TestAccCloud9EnvironmentEC2_tags
=== CONT  TestAccCloud9EnvironmentEC2_allFields
=== CONT  TestAccCloud9EnvironmentEC2_disappears
--- PASS: TestAccCloud9EnvironmentEC2_disappears (159.70s)
--- PASS: TestAccCloud9EnvironmentEC2_basic (160.50s)
--- PASS: TestAccCloud9EnvironmentEC2_allFields (187.30s)
--- PASS: TestAccCloud9EnvironmentEC2_tags (194.91s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/cloud9     197.170s
```

### CODE REVIEW

File: website/docs/cdktf/python/r/cloud9_environment_ec2.html.markdown [hunk 1]
```diff
@@ -30,7 +30,8 @@ class MyConvertedCode(TerraformStack):
         super().__init__(scope, name)
         Cloud9EnvironmentEc2(self, "example",
             instance_type="t2.micro",
-            name="example-env"
+            name="example-env",
```
Comment (JonZeolla) [inline website/docs/cdktf/python/r/cloud9_environment_ec2.html.markdown:33]: I manually changed this and the equivalent from a few `cdktf` files for now so it doesn't get lost.  I think I need to run something to re-generate these files, but I'm not sure what it is?

Comment (ewbankkit) [reply website/docs/cdktf/python/r/cloud9_environment_ec2.html.markdown:33]: The CDKTF documentation periodically gets generated from the `website/docs/[d|r]` files by a GitHub action, it does not need to be manually edited (we are going to update the Contributor Guide to capture this).

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccCloud9EnvironmentEC2_' PKG=cloud9 ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/cloud9/... -v -count 1 -parallel 2  -run=TestAccCloud9EnvironmentEC2_ -timeout 360m
=== RUN   TestAccCloud9EnvironmentEC2_basic
=== PAUSE TestAccCloud9EnvironmentEC2_basic
=== RUN   TestAccCloud9EnvironmentEC2_allFields
=== PAUSE TestAccCloud9EnvironmentEC2_allFields
=== RUN   TestAccCloud9EnvironmentEC2_tags
=== PAUSE TestAccCloud9EnvironmentEC2_tags
=== RUN   TestAccCloud9EnvironmentEC2_disappears
=== PAUSE TestAccCloud9EnvironmentEC2_disappears
=== CONT  TestAccCloud9EnvironmentEC2_basic
=== CONT  TestAccCloud9EnvironmentEC2_tags
--- PASS: TestAccCloud9EnvironmentEC2_basic (135.74s)
=== CONT  TestAccCloud9EnvironmentEC2_disappears
--- PASS: TestAccCloud9EnvironmentEC2_tags (165.17s)
=== CONT  TestAccCloud9EnvironmentEC2_allFields
--- PASS: TestAccCloud9EnvironmentEC2_allFields (162.25s)
--- PASS: TestAccCloud9EnvironmentEC2_disappears (173.57s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/cloud9	317.057s
```

**Comment by ewbankkit** [comment]
@JonZeolla Thanks for the contribution :tada: :clap:.

**Pull request merged**